### PR TITLE
Update VideoPress block for `useBlockProps`

### LINKED ISSUE #18864: Blocks: switch away from __experimentalBlock to useBlockProps
WordPress 5.6 (soon to be required to use the Jetpack plugin) introduces the following change:
https://make.wordpress.org/core/2020/11/18/block-api-version-2/

It will allow us to switch away from `__experimentalBlock` to `useBlockProps` for some of our blocks.

- [ ] Video
- [x] Premium content Take one done https://github.com/Automattic/jetpack/pull/19844, needs take two https://github.com/Automattic/jetpack/pull/19844#issuecomment-844024627

Matching Gutenberg change:
https://github.com/WordPress/gutenberg/pull/31419

*******

Primary issues: #18842  (5.7), https://github.com/Automattic/jetpack/issues/19654 (5.8)

### PR DESCRIPTION
**Pull request by jgcaruso**
Fixes #18864 

#### Changes proposed in this Pull Request:

* extracts the video player portion of the rendered component into a function component so the `useBlockProps` hook can be called

#### Testing instructions:

* Ensure you have a Jetpack site with VideoPress enabled
* Prior to pulling PR, create a post and add a video block (it may be unclickable and very wide, this PR will fix that)
* Apply the PR
* Refresh the post and the videoblock should still work
* Add a new video block, it should render as expected
* Ensure all video block interactions continue to work as expected

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/videopress/edit.js [hunk 1]
```diff
@@ -451,43 +432,70 @@ const VideoPressEdit = CoreVideoEdit =>
...
+						caption={ caption }
+					/>
 				</Fragment>
 			);
 		}
 	};
 
+// The actual, final rendered video player markup
+// In a separate function component so that `useBlockProps` could be called.
+const VpBlock = props => {
```
Comment (roundhill) [inline projects/plugins/jetpack/extensions/blocks/videopress/edit.js:450]: Totally not a blocker, but what if this was `VideoBlock` or `VideoPressBlock`, it just looks a little funny to my eyes.

**Review by roundhill — APPROVED** [review]
Tested fine for me! Looks like we're safe from using deprecation because the props/markup don't change? Old VideoPress blocks still loaded up fine for me.

**Review by jeherve — APPROVED** [review]
This looks good to me. 👍

**Comment by roundhill** [comment]
Thanks @jeherve, @jgcaruso is AFK today so shall I just merge this? Looks like today is the deadline?

**Comment by jeherve** [comment]
Yep, merge away! 🥳

**Comment by jgcaruso** [comment]
Reopened branch to get D61748-code to rebuild

**Pull request merged**